editor/src/components/canvas/controls/insertion-plus-button.tsx

```diff
@@ -52,10 +52,17 @@ export const InsertionControls: React.FunctionComponent<ControlProps> = betterRe
         props.componentMetadata,
       )
       if (child.specialSizeMeasurements.position !== 'absolute' && childFrame != null) {
-        let direction: 'row' | 'column' =
-          child.specialSizeMeasurements.parentLayoutSystem === 'flex'
-            ? parentElement.props?.style?.flexDirection || 'row'
-            : 'column'
+        let direction: 'row' | 'column' = 'column'
+        if (child.specialSizeMeasurements.parentLayoutSystem === 'flex') {
+          switch (child.specialSizeMeasurements.parentFlexDirection) {
+            case 'row':
+            case 'column':
+              direction = child.specialSizeMeasurements.parentFlexDirection
+              break
+            default:
+            // Ignore any other values.
```

Comment:
hmm, how could we safeguard against forgetting to make this work for row-reverse etc? maybe we should bail out from shoing the control, and early return with a `null`

Reply:
Yeah, I was on the fence about that, we could make the `direction` field nullable and skip adding to `controlProps` if it's null.

Reply:
sounds good! also, `controlProps` is probably a _very bad_ variable name because of the loaded meaning coming from the Inspector codebase

Reply:
Fixed in https://github.com/concrete-utopia/utopia/pull/1577/commits/00bb0fb34cfbc91b93ccf9a1d4a4e5485876c460